hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -534,7 +533,7 @@ private class Emit(
           Region.loadIRIntermediate(x.pType)(pArray.elementOffset(xa, len, xi))))
       case ArrayLen(a) =>
         val codeA = emit(a)
-        strict(PContainer.loadLength(coerce[Long](codeA.v)), codeA)
+        strict(PCanonicalArray.loadLength(coerce[Long](codeA.v)), codeA)
```

Comment:
In a followup PR, remove the static method -- this should really be `a.pType.asInstanceOf[PArray].loadLength(...)`.

Save that for a followup PR though.

Reply:
Yep, had already planned this. This PR makes it clear that nothing should rely on the object methods. I don't think the object should exist.